hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -1076,7 +1076,7 @@ extends Importer:
         val sym = members.getOrElse(nme.name, lastWords(s"Symbol not found: ${nme.name}"))
         var newCtx = S(td.symbol).collectFirst:
             case s: InnerSymbol => s
-          .fold(ctx.nestLocal)(ctx.nestInner(_))
+          .fold(ctx.nest(OuterCtx.LambdaOrHandlerBlock))(ctx.nestInner(_))
```

Comment:
It looks like the only time we don't have an inner symbol here would be when we are looking at a type alias (`Als`). Why would you set the context to `LambdaOrHandlerBlock` here? I think we should add a new case to the enum (call it `NonReturnContext`) where using `return` is simply forbidden. Surely we don't want to allow things like `type T = return "lol"`

Reply:
In fact we should add `type T = return "lol"` as a negative test case in `NonLocalReturns.mls`!